File: ApiClases_20240722_Solucion/Frontal/src/app/transaccion/transaccion.ts
export interface ITransaccion {

  constructor(): void;
  id: number;
  idEnvia: number;
  idRecibe: number;
  cantidadEnvia: number;
  cantidadRecibe: number;
  fecha: string;
}

export class Transaccion implements ITransaccion {

    ["constructor"](): void {
        
    }
    id!: number ;
    idEnvia!: number;
    idRecibe!: number;
    cantidadEnvia!: number;
    cantidadRecibe!: number;
    fecha!: string;
}

File: ApiClases_20240722_Solucion/ApiClases_20270722_Proyecto/Entidades/Transaccion.cs

namespace ApiClases_20270722_Proyecto.Entidades;

public class Transaccion
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int IdEnvia { get; set; }

    [Required]
    public double CantidadEnvia { get; set; }

    [Required]
    public int IdRecibe { get; set; }

    [Required]
    public double CantidadRecibe { get; set; }

    [Required]
    public DateTime Fecha { get; set; }

    [Required]
    public string MonedaOrigen { get; set; }

    [Required]
    public string MonedaDestino { get; set; }

    [Required]
    public double CosteTransaccion { get; set; } //Puede ser un valor fijo o un porcentaje. Hay que hablarlo.
}
